src/app/d2-loadout-builder/generated-sets/GeneratedSets.tsx

```diff
@@ -114,7 +137,7 @@ export default class GeneratedSets extends React.Component<Props, State> {
         </select>
 
         <h3>{t('LoadoutBuilder.GeneratedBuilds')}</h3>
-        {matchedSets.map((set, index) => (
+        {matchedSets.slice(0, shownSets).map((set, index) => (
           <div className="generated-build" key={index}>
```

Comment:
You don't want to use `index` as a key - if you generate a new set list, they'll still have the same indexes, and the end result could be some unwanted re-use of the same components.